logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/featureConfig/SyncFeatureConfigsUseCase.kt

```diff
@@ -131,14 +135,24 @@ internal class SyncFeatureConfigsUseCaseImpl(
 
     private fun handleSelfDeletingMessagesStatus(model: SelfDeletingMessagesModel) {
         if (!kaliumConfigs.selfDeletingMessages) {
-            userConfigRepository.setSelfDeletingMessagesStatus(SelfDeletingMessagesStatus(false, null, null))
+            userConfigRepository.setTeamSettingsSelfDeletionStatus(
+                TeamSettingsSelfDeletionStatus(
+                    enforcedSelfDeletionTimer = SelfDeletionTimer.Disabled,
+                    hasFeatureChanged = null
+                )
+            )
         } else {
             val selfDeletingMessagesEnabled = model.status == Status.ENABLED
-            userConfigRepository.setSelfDeletingMessagesStatus(
-                SelfDeletingMessagesStatus(
-                    selfDeletingMessagesEnabled,
-                    null, // when syncing the initial status, we don't know if the status has changed so we set it to null
-                    model.config.enforcedTimeoutSeconds
+            val enforcedTimeout = model.config.enforcedTimeoutSeconds?.toDuration(DurationUnit.SECONDS) ?: ZERO
+            val selfDeletionTimer = when {
+                selfDeletingMessagesEnabled && enforcedTimeout > ZERO -> SelfDeletionTimer.Enforced(enforcedTimeout)
+                selfDeletingMessagesEnabled -> SelfDeletionTimer.Enabled(ZERO)
+                else -> SelfDeletionTimer.Disabled
+            }
+            userConfigRepository.setTeamSettingsSelfDeletionStatus(
+                TeamSettingsSelfDeletionStatus(
+                    enforcedSelfDeletionTimer = selfDeletionTimer,
+                    hasFeatureChanged = null, // when syncing the initial status, we don't know if the status changed so we set it to null
```

Comment:
instead of working with null and not knowing what it means on client side, maybe a simple enum class or a sealed class would be a better idea

Reply:
Therefore I think changing that will imply changing the rest of the flags types, which falls a bit out of the scope for this PR imho.

Reply:
I do not fully agree that it is out of scope of this PR as we can address the other places later on by taking inspiration from you on this PR, but for sake of pushing things forward I am okay with it , minor thing I guess 🦖